app/src/main/java/com/rokt/roktdemo/ui/layouts/ScanQrViewModel.kt

```diff
@@ -54,18 +58,45 @@ class ScanQrViewModel @Inject constructor(
     fun executePreview(embeddedWidget: WeakReference<Widget>?) {
         val scannedData = _state.value.data?.scannedData
         scannedData?.let { data ->
-            val attributes = hashMapOf(
-                ATTRIBUTE_IS_DEMO to true.toString(),
-                ATTRIBUTE_LAYOUT_ID to data.previewId,
-                ATTRIBUTE_VERSION_ID to data.versionId,
-                ATTRIBUTE_CREATIVE_ID to data.creativeIds.joinToString(separator = ",")
-            )
+            val attributes = buildMap {
+                this[ATTRIBUTE_IS_DEMO] = true.toString()
+                this[ATTRIBUTE_LAYOUT_ID] = data.previewId
+                this[ATTRIBUTE_VERSION_ID] = data.versionId
+                this[ATTRIBUTE_CREATIVE_ID] = data.creativeIds.joinToString(separator = ",")
+                data.language?.let { this[ATTRIBUTE_LANGUAGE] = it }
+                getDemoConfig(data)?.let { this[ATTRIBUTE_DEMO_CONFIG] = it }
+            }
             val placeholders = embeddedWidget?.let { hashMapOf(PREVIEW_PLACEHOLDER to it) }
-            roktExecutor.executeRokt("", attributes, placeholders)
+            roktExecutor.executeRokt("", HashMap(attributes), placeholders)
         }
     }
 }
 
+private fun getDemoConfig(previewData: PreviewData): String? {
+    previewData.layoutVariantIds?.let { layoutVariantIds ->
+        val slots = mutableListOf<DemoLayoutSlotConfig>()
+        previewData.creativeIds.forEachIndexed { index, creativeId ->
+            val slot = DemoLayoutSlotConfig(
+                layoutVariantId = layoutVariantIds[index % layoutVariantIds.count()],
```

Comment:
So previewData can have 2 layout variants and 4 creatives and we'll get 4 slots alternating between the variants (I think this is what's in the screenshot)?

Is 4 layout variants and 2 creatives a valid scenario (guessing not)?

Reply:
Yes. Benefits and Text layout variants are the ones supported now. iOS is also implemented the same way.